sample_project/Assets/SampleViewer/Samples/Geometry/Geometries.cs

```diff
@@ -59,29 +59,44 @@ public class Geometries : MonoBehaviour
 	[SerializeField] public Button[] UnitButtons;
 
 	private InputActions inputActions;
+	private TouchControls touchControls;
+	private bool isTouch = false;
 
 	private void Awake()
 	{
 		inputActions = new InputActions();
+		touchControls = new TouchControls();
 	}
 
 	private void OnEnable()
 	{
+#if UNITY_EDITOR && !UNITY_IOS && !UNITY_ANDROID && !UNITY_VISIONOS
 		inputActions.Enable();
 		inputActions.DrawingControls.LeftClick.started += OnLeftClickStart;
 		inputActions.DrawingControls.LeftClick.canceled += OnLeftClickEnd;
 		inputActions.DrawingControls.LeftShift.performed += ctx => OnLeftShift(true);
 		inputActions.DrawingControls.LeftShift.canceled += ctx => OnLeftShift(false);
-	}
+#else
+        touchControls.Enable();
+        touchControls.Touch.TouchPress.started += ctx => OnTouchInputStarted(ctx);
+        touchControls.Touch.TouchPress.canceled += ctx => OnTouchInputEnded(ctx);
+#endif
+    }
 
 	private void OnDisable()
 	{
+#if UNITY_EDITOR && !UNITY_IOS && !UNITY_ANDROID && !UNITY_VISIONOS
 		inputActions.Disable();
 		inputActions.DrawingControls.LeftClick.started -= OnLeftClickStart;
 		inputActions.DrawingControls.LeftClick.canceled -= OnLeftClickEnd;
 		inputActions.DrawingControls.LeftShift.performed -= ctx => OnLeftShift(true);
 		inputActions.DrawingControls.LeftShift.canceled -= ctx => OnLeftShift(false);
-	}
+#else
+        touchControls.Disable();
+        touchControls.Touch.TouchPress.started -= ctx => OnTouchInputStarted(ctx);
```

Comment:
We cant remove lambda expressions this way, it is treated as a different expression. This is done incorrectly with the normal inputActions too.
https://discussions.unity.com/t/unsubscribe-from-an-event-using-a-lambda-expression/882983/2